### ISSUE TITLE
backup files?

### ISSUE THREAD

**Issue by Kirin-kun** [deepfakes/faceswap-playground#275]
I noticed that the training no longer creates backup files regularly for the models. I don't know starting from which commit, but it is fairly recent.

There's a function "backup_files" in utils.py that does the action, but I didn't find it called anywhere in the code.

Is it normal? Or did I miss something?

**Comment by torzdf** [comment]
It now only backs up if the average loss drops between save iterations.

Under the old method, if your model corrupted and you didn't catch it before the next backup, then your backups would corrupt too.

This way, if a model corrupts your backups will be fine.

It is called in `__base.py` in `/plugins/train/model`

**Comment by Kirin-kun** [comment]
I don't know about this. I trained a model for several hours and the average loss went down, but it didn't create any bk files, nor overwrite the old ones.

For both unbalanced and villain models.

**Comment by Kirin-kun** [comment]
It stumped me for a moment, but when reading the debug logs, I think I found out why no new backup files appeared.

After a first round of training, I actually added some faces of the same person to the A face set, reusing the same model files, obviously, so it could further train.

So, the average loss went up, because the model had new data to work on. 

However, now, the average loss is now higher than the historical lowest average. And it will be so until the training reaches lower loss than the previous training with the smaller face set. Only then will it backup my model files. This history is stored in the _state.json file.

So, in order to make backups quickly, if I want to keep the state file, I'd have to edit it and put a ridiculously high value, so on the first save, it will backup everything. 

I guess if I delete the state file, it will recreate one, starting from iteration zero, but it will not reset the model files?

Or, if I may, I'd suggest a feature where one can manually backup the models by pressing a key. Instead of "S' to save immediately, it would be 'B' to save and backup immediately (forcing the "should_backup" to true).

As usual, I'm an edge case :)
